packages/gatsby-theme-docs/src/utils/extract-query-parameters.js

```diff
@@ -0,0 +1,22 @@
+function extractQueryParameters(loc) {
+  const queryString = loc.search.substring(1);
+
+  if (queryString) {
+    const keyValueArray = queryString.split('&');
+    return {
+      fromFilterDate: decodeURIComponent(keyValueArray[0].split('=')[1]),
+      toFilterDate: decodeURIComponent(keyValueArray[1].split('=')[1]),
+      filterTopics: JSON.parse(
+        decodeURIComponent(keyValueArray[2].split('=')[1])
+      ),
+    };
+  }
```

Comment:
No need to parse this manually, I suggest to use the `qss` library and then the `decode` function